src/core/qgsogrutils.cpp

```diff
@@ -1256,6 +1257,18 @@ std::unique_ptr<QgsSymbol> QgsOgrUtils::symbolFromStyleString( const QString &st
     QgsUnitTypes::RenderUnit lineWidthUnit = QgsUnitTypes::RenderMillimeters;
     convertSize( lineStyle.value( QStringLiteral( "w" ) ).toString(), lineWidth, lineWidthUnit );
 
+    // if the pen is a mapinfo pen, use dedicated converter for more accurate results
+    const thread_local QRegularExpression sMapInfoId = QRegularExpression( QStringLiteral( "mapinfo-pen-(\\d+)" ) );
+    const QRegularExpressionMatch match = sMapInfoId.match( string );
+    if ( match.hasMatch() )
+    {
+      const int penId = match.captured( 1 ).toInt();
+      QgsMapInfoSymbolConversionContext context;
+      std::unique_ptr<QgsSymbol> res( QgsMapInfoSymbolConverter::convertLineSymbol( penId, context, color, lineWidth, lineWidthUnit ) );
+      if ( res )
```

Comment:
When can we expect the `res` unique_ptr to be invalid in this situation?

Reply:
There's some symbols we can't reproduce in qgis (zig zag lines), so there's a chance the converter will return nullptr

Reply:
(thanks for the explanation, I didn't see the *zig-zaggy patterns* case)